morpheus/_lib/include/morpheus/messages/control.hpp

```diff
@@ -271,43 +320,189 @@ class ControlMessage
      */
     void task_type(ControlMessageType task_type);
 
+    /**
+     * @brief Sets a timestamp for a specific key within a given group.
+     *
+     * This method stores a nanosecond precision timestamp associated with a unique
+     * identifier composed of a group and key. If the group::key already exists,
+     * its timestamp will be updated to the new value.
+     *
+     * @param key The specific key for which the timestamp is to be set.
+     * @param group The group to which the key belongs, aiding in namespace separation.
+     * @param timestamp_ns The timestamp value in nanoseconds to be associated with the group::key.
+     */
+    void set_timestamp(const std::string& group, const std::string& key, std::size_t timestamp_ns);
```

Comment:
Add overloads that dont require a group and allow users to do the concatenation themselves or to look up without any group.